Tools/LyTestTools/ly_test_tools/o3de/multi_test_framework.py

```diff
@@ -0,0 +1,1383 @@
+        for m in out_matches:
+            try:
+                elem = json.loads(m.groups()[0])
+                found_jsons[elem["name"]] = elem
+            except Exception:  # Intentionally broad to avoid failing if the output data is corrupt
+                logger.warning("Error reading result JSON", exc_info=True)
+                continue
+
+        # Try to find the element in the log output, this is used for cutting the log contents later
+        log_matches = pattern.finditer(log_output)
+        for m in log_matches:
+            try:
+                elem = json.loads(m.groups()[0])
+                if elem["name"] in found_jsons:
+                    found_jsons[elem["name"]]["log_match"] = m
+            except Exception:  # Intentionally broad, to avoid failing if the log data is corrupt
+                logger.warning("Error reading result JSON", exc_info=True)
+                continue
+
+        log_start = 0
+        for test_spec in test_spec_list:
+            name = editor_utils.get_module_filename(test_spec.test_module)
+            if name not in found_jsons.keys():
+                results[test_spec.__name__] = Result.Unknown(
+                    log_attribute,
+                    test_spec,
+                    output,
+                    f"Found no test run information on stdout for {name} in the test output",
+                    log_output)
+            else:
+                result = None
+                json_result = found_jsons[name]
+                json_output = json_result["output"]
+
+                # Cut the log output so it only has the log contents for this run
+                if "log_match" in json_result:
+                    m = json_result["log_match"]
+                    end = m.end() if test_spec != test_spec_list[-1] else -1
+                else:
+                    end = -1
+                cur_log = log_output[log_start: end]
+                log_start = end
+
+                if json_result["success"]:
+                    result = Result.Pass(log_attribute, test_spec, json_output, cur_log)
+                else:
+                    result = Result.Fail(log_attribute, test_spec, json_output, cur_log)
+                results[test_spec.__name__] = result
+
+        return results
+
+    @staticmethod
+    def _report_result(name: str, result: Result.ResultType) -> None:
+        """
+        Raises a pytest failure if the test result is not a PASS, specifying the information
+        :param name: Name of the test
+        :param result: The Result object which denotes if the test passed or not
+        :return: None
+        """
+        if isinstance(result, Result.Pass):
+            output_str = f"Test {name}:\n{str(result)}"
+            print(output_str)
+        else:
+            error_str = f"Test {name}:\n{str(result)}"
+            pytest.fail(error_str)
+
+    def _prepare_asset_processor(self, workspace: AbstractWorkspaceManager, collected_test_data: TestData) -> None:
```

Comment:
This appears to be a static method, and is likely suitable to move into its own function editor_test_utils.py